File: blablamovebackend/src/main/java/fr/polytech/unice/blablamove/teamc/blablamovebackend/model/influxdb/DeliveryIssue.java
package fr.polytech.unice.blablamove.teamc.blablamovebackend.model.influxdb;

import org.influxdb.annotation.Column;
import org.influxdb.annotation.Measurement;

import java.time.Instant;

/**
 * Class DeliveryIssue
 *
 * @author Tanguy Invernizzi
 */
@Measurement(name = "delivery_issue")
public class DeliveryIssue {

    /**
     * The type of the issue.
     */
    @Column(name = "issue_type")
    private String issueType;

    /**
     * The id of this delivery.
     */
    @Column(name = "delivery_uuid")
    private String delivery_uuid;

    /**
     * The time at which the issue happened.
     */
    @Column(name = "time")
    private Instant time;

    public Instant getTime() {
        return time;
    }

    public String getIssueType() {
        return issueType;
    }

    public void setIssueType(String issueType) {
        this.issueType = issueType;
    }

    public String getDelivery_uuid() {
        return delivery_uuid;
    }

    public void setDelivery_uuid(String delivery_uuid) {
        this.delivery_uuid = delivery_uuid;
    }

    public void setTime(Instant time) {
        this.time = time;
    }
}

File: adminMarketingUI/src/app/model/delivery-issue.ts
export class DeliveryIssue {
  private _issueType: string;
  private _delivery_uuid: string;
  private _time: Date;

  constructor(issueType: string, delivery_uuid: string, time: Date) {
    this._delivery_uuid = delivery_uuid;
    this._issueType = issueType;
    this._time = time;
  }

  get issueType(): string {
    return this._issueType;
  }

  set issueType(value: string) {
    this._issueType = value;
  }

  get delivery_uuid(): string {
    return this._delivery_uuid;
  }

  set delivery_uuid(value: string) {
    this._delivery_uuid = value;
  }

  get time(): Date {
    return this._time;
  }

  set time(value: Date) {
    this._time = value;
  }
}
